pkg/ddl/job_worker.go

```diff
@@ -519,9 +519,19 @@ func (w *worker) transitOneJobStep(d *ddlCtx, job *model.Job) (int64, error) {
 		})
 		return 0, w.handleJobDone(d, job, t)
 	}
-	d.mu.RLock()
-	d.mu.hook.OnJobRunBefore(job)
-	d.mu.RUnlock()
+	failpoint.InjectCall("onJobRunBefore", job)
+	// ctcSlowDown is used by failpoint HTTP API in other repo, don't delete it.
+	failpoint.Inject("ctcSlowDown", func() {
```

Comment:
can we enable it with EnableCall("onJobRunBefore") in the api

Reply:
`EnableCall` can only be enabled with code, not environment variables or HTTP API.

Reply:
for this failpoint the logic is fixed, we can handle it differently at the http api side, not here, to make it cleaner

Reply:
OK, I'll try to let `/fail/` HTTP handler call `EnableCall`

Reply:
https://github.com/pingcap/tidb/pull/55215/commits/109c404b7e85d70c7efa239ab74223b341d9f4ba